pkgs/misc/crust/default.nix

```
@@ -0,0 +1,154 @@
+  defaultSrc = fetchFromGitHub {
+    owner = "crust-firmware";
+    repo = "crust";
+    rev = "c308a504853e7fdb47169796c9a832796410ece8";
+    sha256 = "AobVD3Jo/6s0cExHXpYYAqZv/gCplxLlDYVscSCIS6M=";
+  };
+  buildCrust = {
+    version ? "${defaultVersion}"
+  , src ? null
+  , filesToInstall
+  , installDir ? "$out"
+  , defconfig
+  , extraConfig ? ""
+  , extraPatches ? []
+  , extraMakeFlags ? []
+  , extraMeta ? {}
+  , ... } @ args: stdenv.mkDerivation ({
+    # Strip "_defconfig" suffix from 'defconfig' if it's present for clearer output
+    # builtins.match returns 'Null' if the string matches so we have to then check if it's Null
+    # FIXME-QA(Krey): I hate this, but am not aware of better way to write this
+    pname = if (builtins.isNull ((builtins.match "_defconfig$" defconfig)))
+      then "crust-${lib.strings.removeSuffix "_defconfig" defconfig}"
+      else "crust-${defconfig}";
+
+    ###! The package has upstream definitions of configurations for set devices and thus the defconfig has to be always included, beyond that the options are designed to be as flexible as possible to accompany special requirements of various devices
+    ###!
+    ###! Avoid declaring extra configuration (extraConfig) when possible as those should go to the upstream unless you use special usecase device e.g. modded teres-A64
+
+    defconfig = if defconfig == null then
+      throw "crust config not defined!"
+      else defconfig;
+
+    version = if src == null then
+      defaultVersion
+      else version;
+
+    src = if src == null then
+      defaultSrc
+      else src;
+
+    # Only for global crust patches, per-device paches should be included as extras to avoid adding noise which would complicate debugging
+    patches = [] ++ extraPatches;
+
+    postPatch = "";
+
+    nativeBuildInputs = [
+      flex
+      bison
+    ];
+
+    depsBuildBuild = [ buildPackages.stdenv.cc ];
+
+    makeFlags = [
+      # This is double-quotted to prevent string splitting (SC2086)
+      "CROSS_COMPILE=\"${stdenv.cc.targetPrefix}\""
+    ] ++ extraMakeFlags;
+
+    passAsFile = [ "extraConfig" ];
+
+    ###! Phases
+    configurePhase = ''
+      runHook preConfigure
+
+      echo "Configuring for ${defconfig}"
+        make ${defconfig}
+
+      # The variable 'extraConfigPath' is often parsed as blank resulting in a scenario where `cat` is expecting interactive input followed by line-break to continue, this is to reduce the risk of that happening
+      [ -z "$extraConfigPath" ] || {
+        cat $extraConfigPath >> .config
+      }
+
+      runHook postConfigure
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      echo "Creating directory in ${installDir}"
+      mkdir -p "${installDir}"
+
+      # Process 'filesToInstall' from declarations
+      for file in ${lib.concatStringsSep " " filesToInstall}; do
+        cp -v "$file" "${installDir}"
+      done
+
+      runHook postInstall
+    '';
+
```

```suggestion
    installPhase = ''
      runHook preInstall

      mkdir -p "$out"
      cp -t "$out/" build/scp/scp.bin

      runHook postInstall
    '';
```

As said previously, `scp.bin` is the only output.

---

Agree, implemented in upcoming